Add support for Renesas RZ/V2H R8 Core

tiennguyenzg opened the pull request:
Add minimal support for RZ/V2H R8 Core as following:

Interface | Controller | Driver/Component
-- | -- | --
GPIO | on-chip | gpio
Pinctrl | on-chip | pinctrl
UART | on-chip | serial

Need: https://github.com/zephyrproject-rtos/hal_renesas/pull/108
Doc: https://builds.zephyrproject.io/zephyr/pr/91394/docs/boards/renesas/rzv2h_evk/doc/index.html

tiennguyenzg committed:
soc: renesas: Add support for Renesas RZ/V2H R8 core

Add support for Renesas RZ/V2H R8 core

Signed-off-by: Tien Nguyen <tien.nguyen.zg@renesas.com>

tiennguyenzg committed:
dts: arm: renesas: Add support for Renesas RZ/V2H R8 core

Add devicetree to support for Renesas RZ/V2H R8 core

Signed-off-by: Tien Nguyen <tien.nguyen.zg@renesas.com>

tiennguyenzg committed:
boards: renesas: Add support for Renesas RZ/V2H-EVK R8 core

Add support for Renesas RZ/V2H-EVK R8 core

Signed-off-by: Tien Nguyen <tien.nguyen.zg@renesas.com>

tiennguyenzg committed:
tests: drivers: gpio: Add support for RZ/V2H-EVK R8 core

Add support for RZ/V2H-EVK R8 core

Signed-off-by: Tien Nguyen <tien.nguyen.zg@renesas.com>

tiennguyenzg committed:
manifest: Update commit id for hal_renesas

Update commit id for hal_renesas

Signed-off-by: Tien Nguyen <tien.nguyen.zg@renesas.com>

tiennguyenzg committed:
dts: bindings: Add CPU device bindings for Cortex-R8

This commit adds device bindings for Cortex-R8

Signed-off-by: Tien Nguyen <tien.nguyen.zg@renesas.com>

nhutnguyenkc commented on soc/renesas/rz/rzv2h/cr8/soc.c:
```diff
@@ -0,0 +1,76 @@
+#include <zephyr/irq.h>
+#include <zephyr/cache.h>
+
+/* System core clock is set to 800 MHz after reset */
+uint32_t SystemCoreClock BSP_SECTION_EARLY_INIT;
+
```
Porting some fsp irq-related variables is missing

```suggestion
/* System core clock is set to 800 MHz after reset */
uint32_t SystemCoreClock BSP_SECTION_EARLY_INIT;
void * gp_renesas_isr_context[BSP_ICU_VECTOR_MAX_ENTRIES + BSP_CORTEX_VECTOR_TABLE_ENTRIES];
uint16_t g_current_interrupt_num[32];
uint8_t  g_current_interrupt_pointer;
```

nhutnguyenkc commented on soc/renesas/rz/rzv2h/cr8/soc.c:
The same issue for CM33. Please help me add the below to cm33/soc.c

```
void * gp_renesas_isr_context[BSP_ICU_VECTOR_MAX_ENTRIES];

tiennguyenzg committed:
soc: renesas: rz: Update Renesas RZ/V2N, RZ/V2L according to RZ/V2H

Porting gp_renesas_isr_context variable used in FSP interrupt source
to unify with RZ/V2H and prevent build error since they share the same
hal_renesas source code.

Signed-off-by: Tien Nguyen <tien.nguyen.zg@renesas.com>

binhnguyen2434 reviewed (approved):
LGTM!
